src/main/java/seedu/address/logic/parser/AddImportantDateParser.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.logic.commands.AddImportantDateCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.patient.ImportantDate;
+
+/**
+ * Parses the user's input arguments and creates a new AddImportantDate Command
+ */
+public class AddImportantDateParser implements Parser<AddImportantDateCommand> {
+
+    /**
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public AddImportantDateCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DATETIME);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_DATETIME) || argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddImportantDateCommand.MESSAGE_USAGE));
+        }
+
+        Index index;
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (IllegalValueException e) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    AddImportantDateCommand.MESSAGE_USAGE), e);
+        }
+
+        ImportantDate importantDate = ParserUtil.parseImportantDate(argMultimap.getValue(PREFIX_NAME).get(),
+                argMultimap.getValue(PREFIX_DATETIME).get());
+
+        return new AddImportantDateCommand(index, importantDate);
+    }
+
+    /**
+     * Returns true if none of the prefixes contains empty {@code Optional} values in the given
+     * {@code ArgumentMultimap}.
+     */
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
+        return Stream.of(prefixes).allMatch(prefix -> argumentMultimap.getValue(prefix).isPresent());
+    }
```

Comment:
Consider importing the already present method to reduce code duplication.
Could be done by changing access modifier in AddCommandParser or refactoring out the method